CRCTray.cs

```diff
@@ -279,9 +279,9 @@ private void ShowDetailedStatusForm(object sender, EventArgs e)
             statusForm.Focus();
         }
 
-        private void UpdateReceived(StatusResult statusResult)
+        private void StatusReceived(StatusResult statusResult)
         {
-            status.Text = statusResult.CrcStatus;
+            status.Text = statusResult.Success ? statusResult.CrcStatus : InitialState;

```

Comment:
Please check http status code instead. `Success` was a way to implement status code with the binary protocol.

Reply:
Because of the statuscode from the backend, we can not check for the EnsureSuccess.

https://github.com/code-ready/tray-windows/blob/481f700db2388f8b955c5faf71611037363cf4ff/Communication/DaemonCommander.cs#L131-L132

as this would otherwise result in an exception without the body. I am not so sure why there needs to be an inconsistency between the JSON output on the CLI and the HTTP protocol.

The code https://github.com/code-ready/tray-windows/commit/1d9992ab6177775852b94a788c20d95b542873db was actually re-wrapping to get a `Success` and `Error` messaging. So you are suggesting to introduce this workaround?

Reply:
I am not yet challenging the 500 itself, but this "Internal Server Error" is considered from an API as being due to a server-side problem that occurs that is not further specified or un-expected. We are throwing this because the VM is not existing?